test/utils.jl

```diff
@@ -0,0 +1,81 @@
+using ImageMagick
+using Images
+using Test
+
+onfail(body, x) = error("I might have overlooked something: $x")
+onfail(body, _::Test.Pass) = nothing
+onfail(body, _::Tuple{Test.Fail,T}) where {T} = body()
+
+absdiff(x, y) = x > y ? x - y : y - x
+
+function imdiff(a, b)
+    a_view = channelview(a)
+    b_view = channelview(b)
+    diffscale = max(maxabsfinite(a_view), maxabsfinite(b_view))
+    d = sum(absdiff.(a_view, b_view))
```

Comment:
Using `absdiff` changed the results a lot -- we were overflowing the `UInt`s with the previous implementation.

Reply:
Good catch!